### PR TITLE
Allow specifying multi-select workflow parameters

### LINKED ISSUE #12233: Advanced Workflow Input Constraints
Requested during the GCC 2021 by @selten.

"Are there plans for more parameter types to attempt to limit the inputs? I could see integer having some similar options as text."

All the constraints available for tools are already implemented in the client form so it should be as easy as exposing setting those up in the workflow editor and tracking them in backend modules.

### PR DESCRIPTION
**Pull request by mvdbeek**
This just works, what's needed is:

- [x] Only allow multiple=true connections to actual multiple=true inputs
- [x] gxformat2 language to describe this
- [x] test

Closes https://github.com/galaxyproject/galaxy/issues/12233#issuecomment-1578117516 and https://github.com/galaxyproject/galaxy/issues/16091 and https://github.com/galaxyproject/galaxy/issues/17957

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy_test/workflow/multiple_text.gxwf-tests.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+- doc: |
+    Test that workflows can run with multiple text parameter fed into a multi select parameter.
+  job:
+    input:
+      value:
+      - --ex1
+      - ex2
+      - --ex3
```
Comment (davelopez) [inline lib/galaxy_test/workflow/multiple_text.gxwf-tests.yml]: I'm confused by this syntax. If those are list items for `value`, shouldn't they be indented?
```yml
value:
  - --ex1
  - ex2
  - --ex3
```

Comment (mvdbeek) [reply lib/galaxy_test/workflow/multiple_text.gxwf-tests.yml]: That's up to you, both serialise to a list, I think this is a matter of preference

Comment (davelopez) [reply lib/galaxy_test/workflow/multiple_text.gxwf-tests.yml]: I see... ok, I just found it counterintuitive to me :sweat_smile:

Comment (mvdbeek) [reply lib/galaxy_test/workflow/multiple_text.gxwf-tests.yml]: I've rebased this to add the additional indentation

Comment (davelopez) [reply lib/galaxy_test/workflow/multiple_text.gxwf-tests.yml]: Thank you!

**Comment by jmchilton** [comment]
Thank you for adopting the workflow tests so quick. I appreciate it! This looks great.

**Review by jmchilton — APPROVED** [review]

**Pull request merged**